cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -117,10 +117,22 @@ public void createCollection(String dbName, String collectionName, long dataExpi
         // ensure the _id.origin index, if possible
         try {
             if (dataExpiration != 0) {
-                BasicDBObject keys = new BasicDBObject().append("_id.origin", 1);
```

Comment:
The index for expiration and the index for query optimization are independent. Thus, I'd suggest to kept the old part (i.e. the block for `if (dataExpiration != 0)`) as it was and add the creation of the new index after it. And the samefor the raw collection.

Reply:
Good point!. Fixed in 00224e0b20ba7a67130344d53f9a335fdc43c203